src/collections/vector.h

```diff
@@ -13,11 +13,18 @@ class Vec {
   private:
   T* buf;
   size_t len;
+  size_t cap;
 
   public:
   // Get the length of the vector
   size_t size();
 
+  // Get the size of underneath array
+  size_t capacity();
+
+  // Set the size of underneath array
+  size_t capacity(size_t s);
```

Comment:
Why this method is a setter while `size_t capacity();` is a getter. It's really a bad code style. Try to add a method called `reserve` instead. There are two methods you can refer to:
* https://doc.rust-lang.org/std/vec/struct.Vec.html#method.reserve
* https://doc.rust-lang.org/std/vec/struct.Vec.html#method.reserve_exact